user-management/src/main/scala/com/normation/plugins/usermanagement/UserManagementService.scala

```diff
@@ -102,52 +133,48 @@ object UserManagementService {
 
   def computeRoleCoverage(roles: Set[Role], authzs: Set[AuthorizationType]): Option[Set[Role]] = {
 
-    def compareRights(r: Role): Option[Role] = {
-      if (r.name == "no_rights") {
+    def compareRights(r: Role, as: Set[AuthorizationType]): Option[Role] = {
+      if (r == Role.NoRights) {
         None
       } else {
-        val authzNames     = authzs.map(_.id)
-        val roleAuthzNames = r.rights.authorizationTypes.map(_.id)
-        val commonRights   = roleAuthzNames.intersect(authzNames)
+        val commonRights = r.rights.authorizationTypes.intersect(as)
         commonRights match {
           // Intersection is total
-          case cr if cr == roleAuthzNames => Some(r)
-          case cr if cr.nonEmpty          =>
-            val test = cr.flatMap(RoleToRights.parseAuthz)
-            if (test.isEmpty) {
```

Comment:
test can't be empty, there is a check on that in the pattern match (not exactly, but there is no need to parse either, we just have to avoid to `map(_.id)` and remains with the types)